### PR TITLE
remove xdg, use $HOME/.config and $HOME/.cache

### LINKED ISSUE #377: discussion(config): Reconsider if we really want XDG spec
We use [XDG](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html) style to organize our cache and config.

In linux, our cache and config is kept in `~/.cache/envd` and `~/.config/envd`. It works well.

In macOS, our config is kept in `~/Library/Application Support/envd`

I do not think any programmer like the dir name with ` `. `Application Support` is hard to find and type.

We should reconsider if we really want the XDG spec. Or we should just always use `~/.cache/envd` and `~/.config/envd`

### PR DESCRIPTION
**Pull request by kemingy**
Signed-off-by: Keming <kemingyang@tensorchord.ai>

* close #377

### CODE REVIEW

File: pkg/util/fileutil/file.go [hunk 1]
```diff
@@ -82,3 +99,39 @@ func RootDir() (string, error) {
...
+func CacheFile(filename string) (string, error) {
+	if strings.ContainsRune(filename, os.PathSeparator) {
+		return "", fmt.Errorf("filename %s should not contain any path separator", filename)
+	}
+	exist, err := DirExists(DefaultCacheDir)
+	if err != nil {
+		return "", err
+	}
+	if !exist {
+		err = os.Mkdir(DefaultCacheDir, os.ModeDir|0700)
```
Comment (kemingy) [inline pkg/util/fileutil/file.go:131]: align with the implementation of "adrg/xdg"

* https://github.com/adrg/xdg/blob/31e6a328d142d3b8ed43fda69c4f2b55e77d8e79/internal/pathutil/pathutil.go#L42-L60

**Review by gaocegege — APPROVED** [review]

File: pkg/util/fileutil/file.go [hunk 2]
```diff
@@ -17,13 +17,30 @@
...
+	DefaultCacheDir  string
+)
+
+func init() {
+	home, err := os.UserHomeDir()
+	if err != nil {
+		panic(err)
+	}
+	DefaultConfigDir = path.Join(home, ".config/envd")
+	DefaultCacheDir = path.Join(home, ".config/envd")
```
[the remark this answers is not in the document]
Comment (gaocegege) [reply pkg/util/fileutil/file.go:41]: Could you apply the change?

**Comment by gaocegege** [comment]
Hi, is it ready to review/merge?

File: pkg/util/fileutil/file.go (hunk 2 above)
Comment (gaocegege) [inline pkg/util/fileutil/file.go:41]: It is still `.config`. I think it should be `.cache`

Comment (kemingy) [reply pkg/util/fileutil/file.go:41]: Sorry. Rebased the wrong commit.

**Review by gaocegege — APPROVED** [review]
Thanks for your contribution! :tada: :+1:

**Comment by gaocegege** [comment]
Could you please rebase the upstream master?

**Pull request merged**